[py2py3] Migration at level scr/python/A/B/C - slice 2

mapellidario opened the pull request:
Fixes #10098 

#### Status

ready

#### Related PRs

This PR is rebased on `master`.

* Previous migration PR: #10057 (slice1, issue #10055 ). 
* Next migration PR: #10117 (slice3, issue #10114 )

#### Description

Run futurize and some manual changes on the first batch of src/python/A/B/C.

No notable changes or errors have been encountered during this PR.

#### Is it backward compatible (if not, which system it affects?)

It should be. No possible causes for errors have been spot by unit tests. 

#### External dependencies / deployment changes

Requires python-future in both py2 and py3 environments.

mapellidario committed:
[py2py3] src/python/A/B/C slice 2 - issue 10098 - src

mapellidario committed:
[py2py3] src/python/A/B/C slice 2 - issue 10098 - test and pylint

amaltaro commented:
@mapellidario since this PR has been rebased on top of #10057, I'd rather focus on 10057 and get that one merged first. Then you rebase it once again and many changes/files will be gone, making it a much easier review to be done.
Please correct me if I'm missing anything though.

mapellidario commented:
I totally agree! Sorry for the noise, thanks again!

mapellidario commented:
@amaltaro after discussion on slack, this has been rebased directly onto master and it is ready to be review!

amaltaro commented on src/python/WMCore/DAOFactory.py:
```diff
@@ -25,7 +28,7 @@ def __call__(self, classname):
             #TODO: Make good
             dialect = None
-            for i in self.dialects.keys():
+            for i in list(self.dialects):
```
It looks like this one could have been:
```
for i in self.dialects:
```

mapellidario replied:
Uh! yes, thanks!

amaltaro commented on src/python/WMCore/Database/DBCreator.py:
```diff
@@ -112,6 +108,6 @@ def __str__(self):
         string = ''
         for i in self.create, self.constraints, self.inserts, self.indexes:
-            for j in i.keys():
+            for j in i:
```
such a horrible variable naming... nothing for you to do though!

amaltaro reviewed (approved):
These changes look good to me, Dario. I assume they are ready to be merged as well?

amaltaro commented:
Actually, can you please squash commits 2 and 3?

mapellidario commented:
I squashed commit about `test` and `pylint` and added a commit with your suggestion about `WMCore/DAOFactory.py`. 

After you give the ok, I can squash the commits about `src` and then this is ready to be merged

amaltaro commented:
Thanks Dario. Feel free to squash the `src` changes and it's ready to go.

amaltaro commented:
Oups, I overlooked the python2.6 idioms. Have you tried updating those (at the very bottom of the PR report)? Or should we leave it for another PR?

mapellidario commented:
Since it is the usual `type` vs `isisntance` that broke JSONThunker a couple of months ago, i am a bit scared of touching those, especially when `dict`s are involved. I can give it a try and hope for the best, though

amaltaro commented:
@mapellidario Dario, I hope you haven't started working on this type x isinstance thing.
As suggested over slack, how about we deal - globally - with this type vs isinstance change in a different PR. I do not expect many modules to be changed and we can nicely isolate that from anything else, making our life easier to spot issues like the JSONThunker one.
